pages/api/sigil/[address]/tokenId/[tokenId].ts

```diff
@@ -0,0 +1,263 @@
+import { getHashesCollectionContract, getHashesContract } from '../../../../../util';
+import Collections from './collections.json';
+import { isValidAddress, hashType } from '../../../../../util/validate';
+
+//http://localhost:3000/api/sigil/{address}/token/{tokenId}
+//permutation-3.vercel.app
+
+//To-do:
+
+//Custom functions for each of the returned paramaters so that default/customised can be set where necessary
+
+///////////////
+//The Structs//
+///////////////
+
+type ProcessedTokenData = {
+  hash: string
+  phrase_value: string
+  binary_value: string
+  type: hashType
+};
+
+type ProcessedWalletData = {
+  dao: number
+  non_dao: number
+  votes: number
+  proposals: number
+};
+
+/////////////////////////////////
+//The Data Collection Functions//
+/////////////////////////////////
+
+//Gets the Token Hashes data from pre-existing API
+async function getTokenAPIData(tokenId: number) {
+  try {
+    const APIdata = await fetch(`http://permutation-3.vercel.app/api/tokenId/${tokenId}`);
+
+    const data = await APIdata.json();
+
+    return data;
+
+  } catch (error) {
+    console.error(`error calling API for tokenID data: ${error}`);
+  }
+}
+
+//Returns processed data for easy use when constucting the NFT
+function processTokenAPIData(data: any): ProcessedTokenData {
+
+  const processedTokenData = {} as ProcessedTokenData;
+
+  try {
+
+    processedTokenData.hash = data.hash;
+
+    processedTokenData.phrase_value = data.phrase_value;
+
+    processedTokenData.binary_value = data.binary_value;
+
+    processedTokenData.type = data.type;
```

Comment:
Consider destructuring the data object and simply returning the values, without `processedTokenData`, ex:  

```js

const processTokenAPIData: ProcessedTokenData  = ({
hash, 
phrase_value, 
binary_value, 
type
}) => ({
 hash,
 phrase_value,
 binary_value,
 type
});
```